Lib.AspNetCore.ServerSentEvents/ServerSentEventsClientValidationArgs.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.AspNetCore.Http;
+
+namespace Lib.AspNetCore.ServerSentEvents
+{
+    /// <summary>
+    /// Provides data for the <see cref="IServerSentEventsService.ValidationHandler"/> event.
+    /// </summary>
+    public struct ServerSentEventsClientValidationArgs
+    {
+        #region Properties
+        /// <summary>
+        /// Gets the request which has been made in order to establish the connection.
+        /// </summary>
+        public HttpRequest Request { get; }
+
+
+        /// <summary>
+        /// Gets the response, if necessary to handle the request.
+        /// </summary>
+        public HttpResponse Response { get; }
```

Comment:
I don't think that `HttpResponse` shoul dbe available here. The validator should only validate and response is not needed for that. The protocol aspects of what to do with response if validation fails should be handled in middleware. If the intend is to have multple ways of handling failed validation, I would introduce enum instead of bool which tells middlewar what to do (for example bad request vs go away).

Reply:
Maybe "validation" and "failed validation" leads the wrong way. It's basically the same thing you would do with the request, context, query string parameters, etc on any GET request, which can mandate writing a "custom" response. It can be arbitrarily complex, but it doesn't even need to be to be more complex than picking from a list of enum choices.

Reply:
Ok, so this is something which should be handled a little bit differently than what we have been discussing, I'll elaborate in a comment at the end.